### PR TITLE
fix: support volar 1.0

### LINKED ISSUE #3785: Volar doesn't seem to start properly
### Thank you for the bug report

- [X] I am using the latest version of `lsp-mode` related packages.
- [X] I checked [FAQ](https://emacs-lsp.github.io/lsp-mode/page/faq/) and [Troubleshooting](https://emacs-lsp.github.io/lsp-mode/page/troubleshooting/) sections
- [ ] You may also try reproduce the issue using clean environment using the following command: `M-x lsp-start-plain`

### Bug description

Volar doesn't seem to start properly, it shows: `Cannot read properties of undefined (reading 'createLanguageService')"`

### Steps to reproduce

1. Install volar with `lsp-install-server`
2. Open a .vue file

### Expected behavior

It should start properly (without error messages)

### Which Language Server did you use?

volar, ts-ls, eslint

### OS

Linux

### Error callstack

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by jadestrong**
Volar has update the options for the language server. See https://github.com/johnsoncodehk/volar/pull/1916 .  
And fix #3785

### CODE REVIEW

**Comment by yyoncho** [comment]
Can you add changelog entry? That will break the config of those using the old server right?

**Comment by yyoncho** [comment]
Can you resolve the conflicts? Then it is good to go

**Review by yyoncho — APPROVED** [review]

**Comment by yyoncho** [comment]
Thank you!

File: clients/lsp-volar.el [hunk 1]
```diff
@@ -135,74 +111,11 @@ in the WORKSPACE-ROOT."
...
-  :activation-fn 'lsp-volar--activate-p
-  :priority 0
-  :multi-root nil
-  :add-on? t
-  :server-id 'volar-html
+  :server-id 'vue-semantic-server
   :initialization-options (lambda () (ht-merge (lsp-configuration-section "typescript")
-                                               (lsp-configuration-section "documentFeatures")))
+                                               (ht ("serverMode" 0)
+                                                   ("diagnosticMode" 1)
```
Comment (johnsoncodehk) [inline clients/lsp-volar.el:117]: Here are a wrong field: `diagnosticMode` -> `diagnosticModel`

Comment (jadestrong) [reply clients/lsp-volar.el:117]: Thanks for your correction, I will fix it.

**Pull request merged**